canal/sync.go

```diff
@@ -101,19 +102,24 @@ func (c *Canal) runSyncBinlog() error {
 			}
 		case *replication.QueryEvent:
 			// handle alert table query
-			if mb := expAlterTable.FindSubmatch(e.Query); mb != nil {
-				if len(mb[1]) == 0 {
-					mb[1] = e.Schema
+			var mb = [][]byte{}
+			// run alter table directly
+			if mb = expAlterTable.FindSubmatch(e.Query); mb == nil {
```

Comment:
@siddontang I think this extract a function is unnecessary, because the alter table and rename table will do the same things after this precess.